FINERACT-807  -  Use underscore instead of dash in tenants and default tenant database names

xurror opened the pull request:
## Description
The Fineract tenants and tenant database names by default are currently mifosplatform-tenants and mifostenant-default respectively.

The Google Cloud Platform console for managed MySQL (Cloud SQL) prevents creating MySQL databases containing dashes; the UI says Database name: Needs to follow the MySQL identifier rules.

While this limitation appears to only be enforced in the Google Cloud UI (it's still possible to create databases containing '-' dashes using the gcloud CLI), the linked doc does seem to suggest to use only underscores.

To make this simpler, and comply with mySQL doc recommendation, let's change mifosplatform-tenants to fineract_tenants and mifostenant-default to fineract_default. (The change from mifosplatform to fineract is just because this seems a good opportunity to make that change, and because it's probably less and not more confusing to change the entire name, and not just a dash to and underscore.)

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Commit message starts with the issue number from https://issues.apache.org/jira/projects/FINERACT/. Ex: FINERACT-646 Pockets API.

- [ ] Coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions have been followed.

- [ ] API documentation at https://github.com/apache/fineract/blob/develop/api-docs/apiLive.htm has been updated with details of any API changes.

- [ ] Integration tests have been created/updated for verifying the changes made.

- [ ] All Integrations tests are passing with the new commits.

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the list.)

Our guidelines for code reviews is at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide

vorburger reviewed (changes requested):
This PR is somewhat confusing, because it has two commits, the first of which is an old and wrong/bad version of the previously raised and still in review other PR #698, and the second commit then "un-does" tje change the first commit makes to `docker/server.xml` ... this won't make sense to other people who will look at this in the future (or users who want to revert or choose to cherry pick only one of FINERACT-807 or FINERACT-806... could I suggest that we first finish up PR #698, and that you then update this PR to include only FINERACT-807 after that?

xurror commented:
Sure, I've updated #698 , just left to be merge so I can update this one

vorburger commented:
@xurror you could now rebase and clean up this PR?

xurror committed:
FINERACT-807

change mifosplatform-tenants and mifostenant-default to fineract_tenants
 and fineract_default respectively

closes https://issues.apache.org/jira/projects/FINERACT/issues/FINERACT-807?filter=allopenissues

xurror commented:
@vorburger what's lacking to merge this PR?

vorburger commented:
LGTM! @xurror the only thing missing was that I hadn't noticed that you fixed this... :smile: In the future when you fix up a PR (and ideally only after it actually passes the build..) please "tag" (@vorburger) me with a short line such as "fixed and all good now (build passing), could you re-review" kind of comment.
